nce/NCE_MP_Pytorch/model.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+class PairwiseConv(nn.Module):
+    """Pairwise model based on MP-CNN"""
+    def __init__(self, model):
+        super(PairwiseConv, self).__init__()
+        self.convModel = model
+        self.dropout = nn.Dropout(model.dropout)
+        self.linearLayer = nn.Linear(model.n_hidden, 1)
+        self.posModel = self.convModel
+        self.negModel = self.convModel
+
+    def forward(self, input):
+        pos = self.posModel(input[0].sentence_1, input[0].sentence_2, input[0].ext_feats)
+        neg = self.negModel(input[1].sentence_1, input[1].sentence_2, input[1].ext_feats)
+        pos = self.dropout(pos)
+        neg = self.dropout(neg)
+        pos = self.linearLayer(pos)
+        neg = self.linearLayer(neg)
+        combine = torch.cat([pos, neg], 1)
+        return combine
+
+
+class MPCNN(nn.Module):
```

Comment:
This code is still repeated. Please import it.

Reply:
Not really... I delete the final three layers (dropout, linear and LogSoftmax) and forward() outputs a feature matrix with size (N, hidden_layer_units). Maybe I need to add a NCE flag here to select whether I will add these layers.

Reply:
What you can do in this case is you can subclass MP-CNN and just override whatever you need to override. The goal is as little duplicate code as possible.

Reply:
Sure. Thanks

Reply:
Feel free to make changes to MP-CNN model to make the OOP more organized if need be.